modin/core/dataframe/pandas/metadata/dtypes.py

```diff
@@ -497,21 +495,18 @@ def _merge_dtypes(
                 # in the 'dtypes_matrix'
                 series = pandas.Series(dtypes, name=i)
                 dtypes_matrix = pandas.concat([dtypes_matrix, series], axis=1)
-                dtypes_matrix.fillna(
-                    value={
-                        # If we encountered a 'NaN' while 'val' describes all the columns, then
-                        # it means, that the missing columns for this instance will be filled with NaNs (floats),
-                        # otherwise, it may indicate missing columns that this 'val' has no info about,
-                        # meaning that we shouldn't try computing a new dtype for this column,
-                        # so marking it as 'unknown'
-                        i: (
-                            pandas.api.types.pandas_dtype(float)
```

Comment:
IIUC there is no point in filling nans with this value, since this will be done later in the code.